### DISCUSSION TITLE
Open and resize.

### DISCUSSION THREAD

**Discussion by blank-manash** [i3/i3#4930] [Q&A]
Hi, I want to bindsym a key to `open a window and resize it automatically'
I have been trying this but it doesn't work

`bindsym $mod+Shift+t exec telegram-desktop resize shrink width 30 px`

**Comment by Airblader** [comment]
You can use something like this, of course you have to adjust the match criterion so it actually matches the window:

```
bindsym $mod+Shift+t exec telegram-desktop
for_window[class=telegram] resize shrink width 30 px
```

You cannot do this after the `exec` directly because `exec` just executes a command, there's no guarantee that any particular window has opened by the time the next command runs.

**Reply by Airblader** [reply]
Depending on your use case, using layout saving and restoring might be another good alternative. You can find docs for that on the i3 website.

**Reply by blank-manash** [reply]
I have tried `for_window [class="TelegramDesktop"] resize shrink width 30 px` but this doesn't seem to be working. 
Also I think I might need to set a particular size instead of shrink but I don't know the default size of it though.

Thank You Airblader.
